src/test/kotlin/org/rust/lang/core/resolve/RustTypeAwareResolveTestCase.kt

```diff
@@ -25,9 +25,12 @@ class RustTypeAwareResolveTestCase : RustResolveTestCaseBase() {
 
     fun testHiddenInherentImpl() = checkIsBound()
     fun testWrongInherentImpl() = checkIsUnbound()
-    fun testNonInherentImpl() = checkIsBound()
     fun testImplGenericsStripped() = checkIsBound()
 
+    fun testNonInherentImpl1() = checkIsBound()
+    fun testNonInherentImpl2() = checkIsBound()
+    fun testNonInherentImpl3() = checkIsBound()
```

Comment:
Yeah, the example turned out to be a little bit more sophisticated, than i've initially deemed it.